From 48309ca1d04209024fbdc80987e0d40e739ef4b0 Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Tue, 12 Jan 2021 15:21:55 +0100
Subject: [PATCH] zynqmp-topic-miamiplusmp: Move ethernet PHY reset to MDIO
 reset

The MDIO controller first probes the bus for devices and only then
asserts the reset for each one. If the reset is already asserted. the
PHY will completely fail to probe, as the reset won't be de-asserted
before accessing the phy.

To solve this, have the MDIO controller trigger the reset and deassert
it before probing the bus. Thus the reset becomes a bus signal.
---
 arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts
index f09150e2fbc1..089a5ed3fbf3 100644
--- a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts
@@ -388,15 +388,15 @@
 	mdio {
 		#address-cells = <1>;
 		#size-cells = <0>;
+		reset-gpios = <&gpioex 2 1>; /* Active low, on port 2 of GPIO expander */
+		/* Datasheet: 10 ms reset pulse, wait 50 ms for register access */
+		reset-delay-us = <10000>;
+		reset-post-delay-us = <50000>;
 
 		miamiphy: ethernet-phy@1 {
 			/* PHY on the board */
 			device_type = "ethernet-phy";
 			reg = <0x1>;
-			reset-gpios = <&gpioex 2 1>; /* Active low, on port 2 of GPIO expander */
-			/* Datasheet: 10 ms reset pulse, wait 50 ms for register access */
-			reset-assert-us = <10000>;
-			reset-deassert-us = <50000>;
 			interrupt-parent = <&gpio>;
 			interrupts = <27 IRQ_TYPE_LEVEL_LOW>;
 			marvell,reg-init =
-- 
2.17.1

